src/Utilities/ToolLocationHelper.cs

```diff
@@ -2108,24 +2108,22 @@ public static IList<string> GetPathToReferenceAssemblies(string targetFrameworkR
                 }
             }
 
-            lock (s_locker)
-            {
-                s_cachedReferenceAssemblyPaths[referenceAssemblyCacheKey] = dotNetFrameworkReferenceAssemblies;
-            }
-
             for (int i = 0; i < dotNetFrameworkReferenceAssemblies.Count; i++)
             {
                 if (
                     !dotNetFrameworkReferenceAssemblies[i].EndsWith(
                         Path.DirectorySeparatorChar.ToString(),
```

Comment:
There is an unnecessary allocation of a string here. The fix is 
` !dotNetFrameworkReferenceAssemblies[i].EndsWith(Path.DirectorySeparatorChar)`
Which is an ordinal comparison anyway.

Reply:
The `char` overload was added in .NET Core 2.0; it won't compile for the desktop .NET Framework.

Reply:
Oh yes, I see I signed off 😃 

Well it can just do the same thing - check length then use indexer.

Reply:
We have a centralized mechanism--`FileUtilities.EndsWithSlash()`.

Reply:
@rainersigwald `FileUtilities.EnsureTrailingSlash` is even more suitable here, however, both of them slightly change the semantics since they also check for `Path.AltDirectorySeparatorChar` at the end of the path.  Any semantic changes should be done separately.

Reply:
https://github.com/Microsoft/msbuild/issues/3544